lib/internal/util.js

```diff
@@ -212,68 +211,84 @@ function assertTypeScript() {
     throw new ERR_NO_TYPESCRIPT();
 }
 
+const ENCODINGS = {
```

Comment:
The existing `encodingsMap` already is shaped like this:

```
{
    ascii: 0,
    utf8: 1,
    base64: 2,
    utf16le: 3,
    latin1: 4,
    hex: 5,
    buffer: 6,
    base64url: 7
},
```

and is generated from an array automatically created by the C++ layer, which avoids hard-coding. I think it's better to keep using that instead of creating this hard-coded map in JavaScript to prevent it from getting out of sync, reusing it also avoids that ABI-breaking change in node.h, which would make this semver-major.